### PR TITLE
Remove unused variable

### PR DESCRIPTION
**Pull request by pepone**
This is just a minor fix to remove an unused variable, I also enable the "Treat Warnings as Error" option for PromiseKit target.

### CODE REVIEW

**Comment by guillaumealgis** [comment]
I don't necessarily agree with the "Treat Warnings as Error" bit, but fixing the warning for the unused variable is great, thanks :)

**Comment by mxcl** [comment]
Yeah, we will regret treat warnings as errors when swift introduces a new warning and Carthage users cannot compile our package, that has to go I'm afraid.

File: Sources/when.swift [hunk 1]
```diff
@@ -154,14 +154,12 @@ public func when<It: IteratorProtocol>(fulfilled promiseIterator: It, concurrent
         }
         guard shouldDequeue else { return }
 
-        var index: Int!
```
Comment (RomanPodymov) [inline Sources/when.swift:157]: I think it was already fixed in #1154.

File: PromiseKit.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1071,7 +1071,7 @@
 				LD_RUNPATH_SEARCH_PATHS = "$(inherited) @executable_path/Frameworks @loader_path/Frameworks";
 				PRODUCT_BUNDLE_PACKAGE_TYPE = FMWK;
 				SKIP_INSTALL = YES;
-				SWIFT_TREAT_WARNINGS_AS_ERRORS = NO;
+				SWIFT_TREAT_WARNINGS_AS_ERRORS = YES;
```
Comment (RomanPodymov) [inline PromiseKit.xcodeproj/project.pbxproj:1074]: As you can see at https://github.com/mxcl/PromiseKit/pull/1144#issuecomment-637866172 you should keep `SWIFT_TREAT_WARNINGS_AS_ERRORS = NO;`.

**Review by RomanPodymov — COMMENTED** [review]
I think this pull request should be closed.

**Pull request closed without merging**